rust/src/types.rs

```diff
@@ -3292,7 +3292,15 @@ impl ShardsVar {
       if let Err(e) = shard.cleanup() {
         shlog!("Errors during shard cleanup: {}", e);
       }
-      shard.destroy();
+
+      let mut tmp = SHVar {
+        valueType: SHType_ShardRef,
+        payload: SHVarPayload {
+          __bindgen_anon_1: SHVarPayload__bindgen_ty_1 { shardValue: shard.0 },
+        },
+        ..Default::default()
+      };
+      destroyVar(&mut tmp);
```

Comment:
The more I look at this PR the more I'm scared haha.. really ShardsVar should only be used for introspection

Reply:
Yeah it doesn't destroy anymore now (I also didn't expect to find destroy there) just decreases the reference count